src/carlib/family_car.rs

```diff
@@ -1,44 +1,36 @@
 use crate::carlib::car::Car;
-use crate::carlib::car_concept::CarConcept;
-use crate::carlib::car_impl::CarImpl;
 
 // Let's try a different approach with the family car.
+const ACCELERATION: f64 = 0.05; // Quite slow.
 pub struct Familycar {
-    pub imp: CarImpl,
+    //don't need acceleration to be a variable, because it never changes
+    pub name: String,
+    pub speed: f64,
 }
 
 impl Familycar {
-    const ACCELERATION: f64 = 0.05; // Quite slow.
-
-    pub fn new() -> Familycar {
-        let mut obj = Familycar {
-            imp: CarImpl::new(),
-        };
-        obj.imp.acceleration = Familycar::ACCELERATION;
-        obj
+    pub fn new(name: Option<String>) -> Self {
+        Self {
+            name: name.unwrap_or("<unnamed>".to_string()),
+            speed: 0.0,
+        }
     }
 }
 
 // Here we implement the Car trait for the Familycar.
 // This time we use the `delegate!` macro.
 // With this macro we can get rid of quite some boilerplate code.
 impl Car for Familycar {
```

Comment:
Is my observation correct that `accelerate` and `brake` do not need to be implemented because they are taken from the Car trait directly?

Reply:
Yes